pkg/sbom/scanner/scanner.go

```diff
@@ -219,48 +222,48 @@ func (s *Scanner) getImageMetadata(request sbom.ScanRequest) *workloadmeta.Conta
 	if store == nil {
 		_ = log.Errorf("workloadmeta store is not initialized")
 		s.scanQueue.AddRateLimited(request)
-		s.scanQueue.Done(request)
 		return nil
 	}
 	img, err := store.GetImage(request.ID())
 	if err != nil || img == nil {
 		log.Debugf("image metadata not found for image id %s: %s", request.ID(), err)
 		s.scanQueue.Forget(request)
-		s.scanQueue.Done(request)
 		return nil
 	}
 	return img
 }
 
 func (s *Scanner) processScan(ctx context.Context, request sbom.ScanRequest, imgMeta *workloadmeta.ContainerImageMetadata, collector collectors.Collector) {
-	if !s.checkDiskSpace(request, imgMeta, collector) {
-		return
+	result := s.checkDiskSpace(imgMeta, collector)
+	errorType := "disk_space"
+
+	if result == nil {
+		scanContext, cancel := context.WithTimeout(ctx, timeout(collector))
+		defer cancel()
+		result = s.performScan(scanContext, request, collector)
+		errorType = "scan"
 	}
-	scanContext, cancel := context.WithTimeout(ctx, timeout(collector))
-	defer cancel()
-	scanResult := s.performScan(scanContext, request, collector)
-	sendResult(request.ID(), &scanResult, collector)
-	s.handleScanResult(scanResult, request, collector)
+	sendResult(request.ID(), result, collector)
+	s.handleScanResult(result, request, collector, errorType)
 	waitAfterScanIfNecessary(ctx, collector)
 }
 
 // checkDiskSpace checks if there is enough disk space to perform the scan
-// It sends an error result to the collector if there is not enough space
-// It returns a boolean indicating if the scan should be pursued
-func (s *Scanner) checkDiskSpace(request sbom.ScanRequest, imgMeta *workloadmeta.ContainerImageMetadata, collector collectors.Collector) bool {
-	if err := s.enoughDiskSpace(collector.Options()); err != nil {
-		result := sbom.ScanResult{
-			ImgMeta: imgMeta,
-			Error:   fmt.Errorf("failed to check current disk usage: %w", err),
-		}
-		sendResult(request.ID(), &result, collector)
-		telemetry.SBOMFailures.Inc(request.Collector(), request.Type(), "disk_space")
```

Comment:
Telemetry isn't emitted here anymore. It is emitted in the `handleScanResult` func